lib/connection_tester.rb

```diff
@@ -0,0 +1,243 @@
+
+class ConnectionTester
+  REQUEST_OPTS = {
+    open_timeout: 6,
+    ssl_timeout:  6,
+    read_timeout: 12
+  }
+
+  NODEINFO_SCHEMA   = "http://nodeinfo.diaspora.software/ns/schema/1.0"
+  NODEINFO_FRAGMENT = "/.well-known/nodeinfo"
+
+  class << self
+    # Test the reachability of a server by the given HTTP/S URL.
+    # In the first step, a DNS query is performed to check whether the
+    # given name even resolves correctly.
+    # The second step is to send a HTTP request and look at the returned
+    # status code or any returned errors.
+    # This function isn't intended to check for the availability of a
+    # specific page, instead a GET request is sent to the root directory
+    # of the server.
+    # In the third step an attempt is made to determine the software version
+    # used on the server, via the nodeinfo page.
+    #
+    # @param [String] server URL
+    # @return [Result] result object containing information about the
+    #                  server and to what point the connection was successful
+    def check(url)
+      url = "http://#{url}" unless url.include?('://')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.